src/app/games/dvonn/tests/DvonnRules.spec.ts

```diff
@@ -322,8 +314,15 @@ describe('DvonnRules:', () => {
                 [_, _, _, _, _, _, _, _, _, N, N],
             ];
             const state: DvonnState = new DvonnState(board, 0, false);
-            const node: DvonnNode = new MGPNode(null, null, state);
+            const node: DvonnNode = new DvonnNode(state);
             RulesUtils.expectToBeDraw(rules, node, minimaxes);
         });
     });
+    describe('isMovablePiece', () => {
+        it('should fail if the coord is not on the board', () => {
+            // TODO FOR REVIEW: we don't have a generic failure for this, should it not be checked by each game?
```

Comment:
Well, most game check that the coord is in range, I'm not sure how you want to generalise that.

Reply:
ps: constantisy isMovablePiece ?

Reply:
je parle ici de l'erreur INVALID_COORD qui est propre à Dvonn alors qu'elle devrait être générique. J'ai refactorisé isMovablePiece pour faire un assert plutôt qu'une branche, car il devrait jamais être appelé avec -1, -1 (donc c'est OK qu'il throw une assertion error). Refactorisé le test du coup aussi (et pas constantisé car ça devient un toThrow)

Reply:
Je vois pas bien comment généraliser ce test mais oui, faut essayer d'utiliser RulesFailure si on peut ?

Reply:
ticketté